fiat-api/src/main/java/com/netflix/spinnaker/fiat/shared/FiatAccessDeniedExceptionHandler.java

```diff
@@ -73,6 +78,18 @@ private String authorizationFailureMessage(
     return sj.toString();
   }
 
+  private Map<String, String> requestHeaders(HttpServletRequest request) {
+    Map<String, String> headers = new HashMap<>();
+
+    while (request.getHeaderNames() != null && request.getHeaderNames().hasMoreElements()) {
+      String headerKey = request.getHeaderNames().nextElement();
```

Comment:
won't the enumeration be reset every time you call `getHeaderNames()` (or give you a new enumeration in reset state?)

Reply:
Yes, dumb.